File: backend/src/main/java/com/digginroom/digginroom/controller/dto/TrackResponse.java
package com.digginroom.digginroom.controller.dto;

import com.digginroom.digginroom.domain.track.Track;

public record TrackResponse(String title, String artist, String superGenre, String description) {

    public static TrackResponse of(Track track) {
        return new TrackResponse(
                track.getTitle(),
                track.getArtist(),
                track.getSuperGenre().getName(),
                track.getDescription()
        );
    }
}

File: android/app/src/main/java/com/digginroom/digginroom/data/entity/TrackResponse.kt
package com.digginroom.digginroom.data.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class TrackResponse(
    @SerialName("title")
    val title: String,
    @SerialName("artist")
    val artist: String,
    @SerialName("superGenre")
    val superGenre: String,
    @SerialName("description")
    val description: String
)
